### PR TITLE
TF-9201 Update OAuth Clients Documentation

### PR DESCRIPTION
**Pull request by Netra2104**
### What
Product Documentation and API documentation changes for new Projects: Restrict VCS Connections feature including:

- changelog.mdx: with all the new changes
- oauth-clients.mdx: there is a new relationship that oauth clients has with `projects`. There is a new attribute `organization-scoped` that oauth client has.
- updates to existing VCS product documentation to cover new setting.

### Why
Users can now scope OAuth client connections either globally within the organization or to specific projects.

[PRD](https://docs.google.com/document/d/1oHeuT2wWkOiQbqd2HoF0dL2JGsaZUMHLjh8Q9QL79eU/edit)
[RFC](https://docs.google.com/document/d/1AGPERUek0Zey1EqkG558j1q8Jd_qgL2ZX0y3s5GZkU4/edit)
Demo Video: https://drive.google.com/file/d/1OeZXnIzu2lmB906_Dxp6XXKsXw-9V5ES/view?usp=drive_link

### Screenshots

Screenshot 2023-11-22 at 10 49 59 AM
Screenshot 2023-11-22 at 10 50 20 AM

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 1]
```diff
@@ -104,6 +105,11 @@ curl \
             "related": "/api/v2/organizations/my-organization"
           }
         },
+        "projects": {
+          "data": [
+            { "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }
+          ]
+        },
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:112]: Maybe I'm misunderstanding something, but shouldn't it be invalid to have both `"organization-scoped": true` and `"projects"` set? The [intro sentence](https://github.com/hashicorp/terraform-docs-common/pull/503/files#diff-78cfd284f7cfbdd8eb861d7282b40c9e950a1b84ab6eed93b4c7cb78d22fb077L41) above says "or", for example.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 2]
```diff
@@ -168,6 +175,11 @@ curl \
           "related": "/api/v2/organizations/my-organization"
         }
       },
+      "projects": {
+        "data": [
+          { "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }
+        ]
+      },
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:182]: Same comment as [above](https://github.com/hashicorp/terraform-docs-common/pull/503/files#diff-78cfd284f7cfbdd8eb861d7282b40c9e950a1b84ab6eed93b4c7cb78d22fb077R108).

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 3]
```diff
@@ -218,11 +230,13 @@ Properties without a default value are required.
...
 | `data.attributes.key`                | string |         | The OAuth Client key. It can refer to a Consumer Key, Application Key, or another type of client key for the VCS provider.                                                                     |
 | `data.attributes.http-url`           | string |         | The homepage of your VCS provider (e.g. `"https://github.com"` or `"https://ghe.example.com"` or `"https://gitlab.com"`).                                                                      |
-| `data.attributes.api-url`            | string |         | The base URL as per your VCS provider's API documentation (e.g. `"https://api.github.com"`, `"https://ghe.example.com/api/v3"` or `"https://gitlab.com/api/v4"`).                            |
+| `data.attributes.api-url`            | string |         | The base URL as per your VCS provider's API documentation (e.g. `"https://api.github.com"`, `"https://ghe.example.com/api/v3"` or `"https://gitlab.com/api/v4"`).                              |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:233]: looks like an existing typo with the dangling `or..`, but can we go ahead and fix it in this PR?

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 4]
```diff
@@ -218,11 +230,13 @@ Properties without a default value are required.
...
 | `data.attributes.secret`             | string |         | The OAuth Client secret. For BitBucket Server, this secret is the the text of the SSH private key associated with your BitBucket Server Application Link.                                      |
 | `data.attributes.rsa-public-key`     | string |         | **Required for BitBucket Server in conjunction with the `secret`. Not used for any other providers.** The text of the SSH public key associated with your BitBucket Server Application Link.   |
+| `data.attributes.organization-scoped` | boolean |         | Whether or not the oauth client is scoped to all projects and workspaces in the organization. Defaults to `true`.
+| `data.relationships.projects.data[]` | array\[object] | `[]`      | A list of resource identifier objects that defines which projects are associated with the oauth client. These objects must contain `id` and `type` properties, and the `type` property must be `projects`. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:239]: Assuming I'm not wrong about this, can we be explicit that `data.relationships.projects.data[]` is only used when `data.attributes.organization-scoped` is set to `false`?

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 5]
```diff
@@ -235,6 +249,13 @@ Properties without a default value are required.
       "http-url": "https://github.com",
       "api-url": "https://api.github.com",
       "oauth-token-string": "4306823352f0009d0ed81f1b654ac17a"
+    },
+    "relationships": {
+      "projects": {
+        "data": [
+          { "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }
+        ]
+      }
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:258]: Would this example include `"organization-scoped": false`, or no?

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 6]
```diff
@@ -280,6 +302,11 @@ curl \
           "related": "/api/v2/organizations/my-organization"
         }
       },
+      "projects": {
+        "data": [
+          { "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }
+        ]
+      },
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:309]: Same question as above. :)

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 7]
```diff
@@ -412,10 +455,104 @@ curl \
   https://app.terraform.io/api/v2/oauth-clients/oc-XKFwG6ggfA9n7t1K
 ```
 
+## Attach an OAuth Client to projects
+
+`POST /oauth-clients/:id/relationships/projects`
+
+| Parameter | Description                                                                                        |
+| --------- | -------------------------------------------------------------------------------------------------- |
+| `:id`     | The ID of the oauth client to attach to projects. Use the "List OAuth Clients" endpoint to find IDs. |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:464]:
```suggestion
| `:id`     | The ID of the OAuth client to attach to projects. Use the "List OAuth Clients" endpoint to find IDs. |
```

**Review by robin-norwood — CHANGES_REQUESTED** [review]
Looks good to me, just a couple of capitalization bits, and a question about how `organization-scoped` and `projects` work together.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 4 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx]: I added a sentence.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 6 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:310]: Done.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 8]
```diff
@@ -156,7 +162,8 @@ curl \
       "http-url": "https://github.com",
       "api-url": "https://api.github.com",
       "key": null,
-      "rsa-public-key": null
+      "rsa-public-key": null,
+      "organization-scoped": false
```
Comment (Netra2104) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:166]: @robin-norwood I went back and checked with the team. It is actually possible for organization-scoped to be true and for there to be projects in the relationships too. 
Here is an example of when that could happen:
1. oauth-client is initially organization-scoped
2. The user adds Project A to the oauth-client and now the organization-scoped is false
3. The user sets provider back to organization-scoped, organization-scoped is now true again, **but** we want to retain all the projects that the user added to the oauth-client incase they want it to be project scoped again and so they dont have to re-add all the projects.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 9]
```diff
@@ -218,11 +230,13 @@ Properties without a default value are required.
...
 | `data.attributes.secret`             | string |         | The OAuth Client secret. For BitBucket Server, this secret is the the text of the SSH private key associated with your BitBucket Server Application Link.                                      |
 | `data.attributes.rsa-public-key`     | string |         | **Required for BitBucket Server in conjunction with the `secret`. Not used for any other providers.** The text of the SSH public key associated with your BitBucket Server Application Link.   |
+| `data.attributes.organization-scoped` | boolean |         | Whether or not the oauth client is scoped to all projects and workspaces in the organization. Defaults to `true`.
+| `data.relationships.projects.data[]` | array\[object] | `[]`      | A list of resource identifier objects that defines which projects are associated with the oauth client. This is only used when `data.attributes.organization-scoped` is set to false. These objects must contain `id` and `type` properties, and the `type` property must be `projects`. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
```
Comment (Netra2104) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:239]: @robin-norwood based on my comment above, I would need to remove this sentence
```This is only used when `data.attributes.organization-scoped` is set to false.```

Comment (robin-norwood) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:239]: Maybe reword it like:
```If `data.attributes.organization-scoped` is set to `false`, the OAuth client will only be available to the projects in this list.```

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 8 above)
Comment (robin-norwood) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:166]: Ah, got it. So, if I understand it correctly, if organization-scoped is true, the oauth client will be available to all projects in the org. If it is false, it will only be available to the projects in the `projects` list.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 9 above)
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:239]: Ok, how about:
```suggestion
| `data.attributes.organization-scoped` | boolean |         | Whether or not the oauth client is scoped to all projects and workspaces in the organization. Defaults to `true`.
| `data.relationships.projects.data[]` | array\[object] | `[]`      | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
```

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 8 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:166]: yup exactly

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 10]
```diff
@@ -318,6 +346,8 @@ This PATCH endpoint requires a JSON object with the following properties as a re
...
 | `data.attributes.secret`         | string | (previous value) | The OAuth Client secret. For BitBucket Server, this secret is the the text of the SSH private key associated with your BitBucket Server Application Link.                                    |
 | `data.attributes.rsa-public-key` | string | (previous value) | **Required for BitBucket Server in conjunction with the `secret`. Not used for any other providers.** The text of the SSH public key associated with your BitBucket Server Application Link. |
+| `data.attributes.organization-scoped` | boolean | (previous value) | Whether or not the oauth client is scoped to all projects and workspaces in the organization. Defaults to `true`.                                                                      |
+| `data.relationships.projects`    | array\[object] | (previous value) | An array of references to projects that the oauth client should be assigned to. Sending an empty array clears all project assignments.                                               |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:350]:
```suggestion
| `data.attributes.organization-scoped` | boolean | (previous value) | Whether or not the oauth client is available to all projects and workspaces in the organization.                                                                      |
| `data.relationships.projects`    | array\[object] | (previous value) | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. Sending an empty array clears all project assignments.                                               |
```

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 11]
```diff
@@ -412,10 +456,104 @@ curl \
   https://app.terraform.io/api/v2/oauth-clients/oc-XKFwG6ggfA9n7t1K
 ```
 
+## Attach an OAuth Client to projects
+
+`POST /oauth-clients/:id/relationships/projects`
+
+| Parameter | Description                                                                                        |
+| --------- | -------------------------------------------------------------------------------------------------- |
+| `:id`     | The ID of the oauth client to attach to projects. Use the "List OAuth Clients" endpoint to find IDs. |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:465]:
```suggestion
| `:id`     | The ID of the OAuth client to attach to projects. Use the "List OAuth Clients" endpoint to find IDs. |
```

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 12]
```diff
@@ -412,10 +456,104 @@ curl \
...
+
+### Request Body
+
+This POST endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path | Type           | Default | Description                                                                                                                                                                                                                                                          |
+| -------- | -------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `data[]` | array\[object] |         | A list of resource identifier objects that defines which projects to attach the oauth client to. These objects must contain `id` and `type` properties, and the `type` property must be `projects` (e.g. `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`).    |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:481]:
```suggestion
| `data[]` | array\[object] |         | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`.    |
```

Comment (robin-norwood) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:481]: Also there's no default value given here, while above `[]` is given as the default value. Should it be `[]` here as well?

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 13]
```diff
@@ -412,10 +456,104 @@ curl \
...
+
+### Request Body
+
+This DELETE endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path | Type           | Default | Description                                                                                                                                                                                                                                                            |
+| -------- | -------------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `data[]` | array\[object] |         | A list of resource identifier objects that defines the projects the oauth client will be detached from. These objects must contain `id` and `type` properties, and the `type` property must be `projects`. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:527]:
```suggestion
| `data[]` | array\[object] |         | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
```

Also same question about whether the default value should be `[]` here as well.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 14]
```diff
@@ -1,7 +1,7 @@
 ---
 page_title: OAuth Clients - API Docs - Terraform Cloud
 description: >-
-  Use the `/oauth-clients` endpoint to manage OAuth connections between your organization and a VCS provider. List, show, create, update, and destroy clients using the HTTP API.
+  Use the `/oauth-clients` endpoint to manage OAuth connections between your organization and a VCS provider. List, show, create, update, and destroy clients using the HTTP API. Attach and detach oauth clients to projects.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:4]:
```suggestion
  Use the `/oauth-clients` endpoint to manage OAuth connections between your organization and a VCS provider. List, show, create, update, and destroy clients using the HTTP API. Attach and detach OAuth clients to projects.
```

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 12 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx]: I have added default value for both attach and detach.

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx (hunk 9 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx]: done!

**Review by robin-norwood — APPROVED** [review]

File: website/docs/cloud-docs/vcs/github.mdx [hunk 15]
```diff
@@ -90,23 +90,31 @@ Leave the page open in a browser tab. In the next step you will copy values from
 
 1. Click the green "Authorize `<GITHUB USER>`" button at the bottom of the authorization page. GitHub might request your password to confirm the operation.
```
Comment (shernandez5) [inline website/docs/cloud-docs/vcs/github.mdx:92]: @robin-norwood I plan to make these same changes to the following files in addition to this one: github-enterprise.mdx, gitlab-com.mdx, bitbucket-cloud.mdx, azure-devops-services.mdx. For the following: gitlab-eece.mdx, bitbucket-server.mdx, azure-devops-server.mdx the SSH keypair section is slightly different so there will be some nuances between the changes (i.e. step numbers are different)

But hoping you can review the changes I've made here first and then once it looks good to you I can copy those changes to the remainder of files.

File: website/docs/cloud-docs/vcs/github.mdx [hunk 16]
```diff
@@ -90,23 +90,31 @@ Leave the page open in a browser tab. In the next step you will copy values from
...
 
-## Step 4: On Terraform Cloud, Set Up SSH Keypair (Optional)
+## Step 4: On Terraform Cloud, Configure Advanced Settings (Optional)
 
-Most organizations will not need to add an SSH private key. However, if the organization repositories include Git submodules that can only be accessed via SSH, an SSH key can be added along with the OAuth credentials. You can add or update the SSH private key at a later time.
+The settings in this section are optional. The Advanced Settings you can configure are:
+- **Scope of VCS Provider** - You can configure which workspaces can use repositories from this VCS provider. By default the **All Projects** option is selected, meaning this VCS provider is enabled to be used by all workspaces in the organization.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/vcs/github.mdx:96]:
```suggestion
- **Scope of VCS Provider** - You can configure which workspaces can use repositories from this VCS provider. By default the **All Projects** option is selected, meaning this VCS provider is available to be used by all workspaces in the organization.
```

File: website/docs/cloud-docs/vcs/github.mdx [hunk 17]
```diff
@@ -90,23 +90,31 @@ Leave the page open in a browser tab. In the next step you will copy values from
...
+### If You Don't Need to Configure Advanced Settings
 
-- SSH will only be used to clone Git submodules. All other Git operations will still use HTTPS.
-- Do not use your personal SSH key to connect Terraform Cloud and GitHub; generate a new one or use an existing key reserved for service access.
-- In the following steps, you must provide Terraform Cloud with the private key. Although Terraform Cloud does not display the text of the key to users after it is entered, it retains it and will use it for authenticating to GitHub.
-- **Protect this private key carefully.** It can push code to the repositories you use to manage your infrastructure. Take note of your organization's policies for protecting important credentials and be sure to follow them.
+1. Click the "Skip and Finish" button. This returns you to Terraform Cloud's VCS Provider page, which now includes your new GitHub client.
 
-### If You Don't Need an SSH Keypair:
+### If You Do Need to Limit the Scope of this VCS Provider:
```
Comment (robin-norwood) [inline website/docs/cloud-docs/vcs/github.mdx:103]:
```suggestion
### If You Need to Limit the Scope of this VCS Provider:
```

File: website/docs/cloud-docs/vcs/github.mdx [hunk 18]
```diff
@@ -90,23 +90,31 @@ Leave the page open in a browser tab. In the next step you will copy values from
...
+
+1. Click the **Update VCS Provider** button to save your selections.
 
 ### If You Do Need an SSH Keypair:
 
+#### Important Notes
+
+- SSH will only be used to clone Git submodules. All other Git operations will still use HTTPS.
+- Do not use your personal SSH key to connect Terraform Cloud and GitHub; generate a new one or use an existing key reserved for service access.
+- In the following steps, you must provide Terraform Cloud with the private key. Although Terraform Cloud does not display the text of the key to users after it is entered, it retains it and will use it for authenticating to GitHub.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/vcs/github.mdx:115]:
```suggestion
- In the following steps, you must provide Terraform Cloud with the private key. Although Terraform Cloud does not display the text of the key to users after it is entered, it retains it and will use it when authenticating to GitHub.
```

File: website/docs/cloud-docs/vcs/index.mdx [hunk 19]
```diff
@@ -77,6 +77,10 @@ To add an SSH key to a VCS connection, finish configuring OAuth in the organizat
 
 If your infrastructure code is spread across multiple VCS providers, you can configure multiple VCS connections. You can choose which VCS connection to use whenever you create a new workspace.
 
+#### Scoping VCS Connections using Projects
+
+You can configure which workspaces can use repositories from a VCS connection. By default each VCS connection is enabled to be used by all workspaces in the organization. If you need to limit which workspaces can use repositories from a given VCS connection you can change this setting to only be enabled for workspaces in selected projects.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/vcs/index.mdx:82]:
```suggestion
You can configure which projects can use repositories from a VCS connection. By default each VCS connection is enabled for all workspaces in the organization. If you need to limit which projects can use repositories from a given VCS connection, you can change this setting to enable the connection for only workspaces in the selected projects.
```

**Review by robin-norwood — APPROVED** [review]

File: website/docs/cloud-docs/vcs/github-enterprise.mdx [hunk 20]
```diff
@@ -95,30 +95,38 @@ Leave the page open in a browser tab. In the next step you will copy values from
...
+### If You Need to Limit the Scope of this VCS Provider:
 
-1. Click the "Skip and Finish" button. This returns you to Terraform Cloud's VCS Providers page, which now includes your new GitHub Enterprise client.
+1. Select the **Selected Projects** option and use the text field that appears to search for and select projects to enable. All current and future workspaces for any selected projects can use repositories from this VCS Provider. 
+
+1. Click the **Update VCS Provider** button to save your selections.
 
 ### If You Do Need an SSH Keypair:
 
-1. On a secure workstation, create an SSH keypair that Terraform Cloud can use to connect to Bitbucket Cloud. The exact command depends on your OS, but is usually something like:
```
Comment (shernandez5) [inline website/docs/cloud-docs/vcs/github-enterprise.mdx:115]: Updated this reference (line 123 in updated version) to be Github Enterprise and not Bitbucket Cloud. I'm pretty sure that was a mistake : )

File: website/docs/cloud-docs/vcs/azure-devops-server.mdx [hunk 21]
```diff
@@ -53,7 +53,17 @@ Leave the page open in a browser tab. In the next step you will copy values from
 
 1. On the "Configure settings" step there is one textbox. Enter your Azure DevOps Server **Personal Access Token** from Step 2. Click the "Continue" button to continue to the next step.
 
-## Step 4: On Workstation, Create an SSH Key for Terraform Cloud
+## Step 4: On Terraform Cloud, Configure VCS Provider Scope (Optional)
```
Comment (shernandez5) [inline website/docs/cloud-docs/vcs/azure-devops-server.mdx:56]: Adding a new step in before mandatory configuration of SSH Key for ADO Server

File: website/docs/cloud-docs/vcs/bitbucket-server.mdx [hunk 22]
```diff
@@ -110,9 +110,19 @@ Leave the page open in a browser tab, and remain logged in as an admin user.
 
    If this results in a 500 error, it usually means Terraform Cloud was unable to reach your Bitbucket Server instance.
 
-1. Click the "Allow" button. This returns you to Terraform Cloud to enter a SSH key.
+1. Click the "Allow" button. This returns you to Terraform Cloud to configure Advanced Settings.
 
-## Step 3: On Workstation: Create an SSH Key for Terraform Cloud
+## Step 3: On Terraform Cloud, Configure VCS Provider Scope (Optional)
```
Comment (shernandez5) [inline website/docs/cloud-docs/vcs/bitbucket-server.mdx:115]: Adding a new step in before mandatory configuration of SSH Key for Btibucket Server

**Review by shernandez5 — COMMENTED** [review]
@robin-norwood I updated the remainder of the VCS providers to have the same changes. The two that were a little different from the rest were azure-devops-server.mdx and bitbucker-server.mdx. Those might need a quick lookover to make sure I did the right thing there. The rest were just a simple copy and paste (and just needed to insert the correct provider name like Bitbucket Cloud or GitLab.com)

**Review by robin-norwood — APPROVED** [review]
lgtm!
